ext/RMagick/rmutil.c

```diff
@@ -1063,7 +1117,15 @@ void rm_sync_image_options(Image *image, Info *info)
 
     if (info->colorspace != UndefinedColorspace)
     {
+#if defined(IMAGEMAGICK_7)
+        exception = AcquireExceptionInfo();
+        SetImageColorspace(image, info->colorspace, exception);
+        // We should not throw an exception in this method and that is
```

Comment:
Why do we not want to throw an exception here?

Reply:
Just pushed a patch to make it more clear. Does it answer your question?

Reply:
I see. Yep, makes sense!

Reply:
Not for this PR, but should there be a way to transmit the exception back up? Like maybe the caller should pass in an exception reference or something?

Reply:
We could probably fix this without breaking the public PR. But that should be something for a new PR.